### PR TITLE
Bump facia cache age

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?

Increases the `max-age`value in the `Cache-Control` header for all fronts (and rss/json variations).  The value has been bumped from 60 seconds to 5 minutes.

Note that the `CacheTime` objects replace calls to `Cached` with both a `Page` and an `Int`.  I believe this is safe because all the calls with `Page` would default to `CacheTime.Default` previously (see `PressedPage`).  If you have any suggestions for how we could tidy this up further I am open to suggestions.

This PR also introduces a _not found_ cache age which will be overwritten by fastly anyway (and if code comments are to be believed not passed by nginx).  Previously the value was hardcoded at 60 seconds in each call to `Cached` in the `FaciaController`.  It makes no difference other than being a bit tidier.

I also removed the redundant `EditionalisedKey`.
## What is the value of this and can you measure success?

Now that the facia purger lambda is running we can safely turn down the cache age, as fronts will be purged immediately on changes.  This will save us bandwidth especially on fronts and at times that see fewer updates (but consistent page views).
## Does this affect other platforms - Amp, Apps, etc?

Not that I am aware of.
## Request for comment

@johnduffell @TBonnin

### CODE REVIEW

File: facia/app/controllers/FaciaController.scala [hunk 1]
```diff
@@ -97,14 +95,13 @@ trait FaciaController extends Controller with Logging with ExecutionContexts wit
 
   def redirectTo(path: String)(implicit request: RequestHeader): Future[Result] = successful {
     val params = request.rawQueryStringOption.map(q => s"?$q").getOrElse("")
-    Cached(60)(WithoutRevalidationResult(Found(LinkTo(s"/$path$params"))))
+    Cached(CacheTime.Facia)(WithoutRevalidationResult(Found(LinkTo(s"/$path$params"))))
```
Comment (johnduffell) [inline facia/app/controllers/FaciaController.scala]: although in reality, redirects aren't likely to change often so could be cached much longer - facia (and article) already return an excessive number of redirects

File: common/app/model/Cached.scala [hunk 2]
```diff
@@ -21,6 +21,8 @@ object CacheTime {
   object Default extends CacheTime(60)
   object LiveBlogActive extends CacheTime(5)
   object RecentlyUpdated extends CacheTime(60)
+  object Facia extends CacheTime(300)
```
Comment (johnduffell) [inline common/app/model/Cached.scala]: didn't go for the switch in the end? ;)

**Comment by johnduffell** [comment]
yep certainly a neater way of doing it, good use of named variables 
👍 📉

**Comment by jfsoul** [comment]
Going to keep an eye on the effect of this change but early signs are good:

(Blue is Facia NetworkIn, green Facia CPU and orange Facia NetworkOut, deploy was made where the cursor is hovering)

cloudwatch day one

**Comment by TBonnin** [comment]
Very nice 👏 Great job
I wonder why NetworkOut doesn't fall as much as NetworkIn

**Comment by jfsoul** [comment]
NetworkOut has already been optimised by @johnduffell's work to send 304 if front is unchanged?

**Comment by johnduffell** [comment]
I still think it's worth trying for smaller machines as a next step, what do other people think? saving would be a few hundred a month based on jon's figures yesterday

**Comment by jfsoul** [comment]
Update on CPU after this change.  The initial drop has been maintained since deploy, bringing up the question of whether we should drop instance size.  Second graph below shows free physical memory which varies between ~5-7GB, which suggests we might not want to drop down to <15GB total memory.  A switch to memory optimised instances might suit.

image

image

**Comment by johnduffell** [comment]
fantastic, that's really interesting.  you can look at the heap usage as well if you like, we send some custom metrics under frontend-facia name
I'd definitely give a +1 for a trial of shrinking the instances, the savings would be significant

**Pull request merged**